lib/pbench/server/api/__init__.py

```diff
@@ -84,6 +90,27 @@ def register_endpoints(api, app, config):
         resource_class_args=(logger, token_auth),
     )
 
+    api.add_resource(
+        CreateMetadata,
+        f"{base_uri}/metadata",
+        resource_class_args=(config, logger, token_auth),
+    )
+    api.add_resource(
+        GetMetadata,
+        f"{base_uri}/metadata/<string:key>",
+        resource_class_args=(config, logger, token_auth),
+    )
+    api.add_resource(
+        QueryMetadata,
```

Comment:
Just wondering what's the difference between "GetMetadata" and "QueryMetadata"

Reply:
The `GetMetadata` returns _all_ metadata values for the authorized user, while `QueryMetadata` specifies a metadata `id` and returns just that value.

Having two classes somewhat obscures the relationship, but if you look at the URI registration in `__init__.py` you'll see that they're both `GET /metadata` with effectively an optional `id` ... that is, `GetMetadata` maps to just `GET /metadata` while `QueryMetadata` maps to `GET /metadata/<id>`. I think Nikhil looked into combining those and found that Flask didn't provide a straightforward way to make the `<id>` parameter optional; the pattern wouldn't match if it was omitted, so he had to add the second class to make it work either way.